packages/nimble-components/src/table/styles.ts

```diff
@@ -25,10 +26,24 @@ export const styles = css`
 
     :host {
         height: 480px;
+        ${
+            /**
+             * Set a default maximum height for the table of 250 rows (including the header row) so
+             * that clients don't accidentally create a table that tries to render too many rows at once.
+             * If needed, the max-height can be overridden by the client, but setting a default ensures
+             * that the max-height is considered if a larger one is needed rather than being overlooked.
+             */ ''
+        }
+        max-height: calc(250 * ${controlHeight});
```

Comment:
Stepping back a bit, the use case we are really trying to help here is compact displaying of relatively small tables that fit in the viewport.

We aren't currently helping super long tables that would go outside the viewport and rely on some other container for scrolling. To support them we should have a mode where the intersection observer is tracking the whole viewport so that can be efficiently virtualized.

I'd say lets cap it at like 40 (i.e. number of rows that fit on my desktop landscape viewport). Something like the icon-table can just set `max-height: none;` to let it grow and handle it's own consequences (it wouldn't want to be virtualized anyway since it's goal is letting ctrl+f page search happen).

Reply:
I'd be fine capping it at 40 rows by default.

Reply:
Other bit from talking offline is maybe we do a similar thing as combobox / select and have a half height offset, i.e. 40.5 so clearer to confirm additional rows are there